examples/Huggingface_Transformers/Image_classification_docker/README.md

```diff
@@ -0,0 +1,165 @@
+framework="pt"       
+repoUrl="https://huggingface.co/apple/mobilevit-xx-small"
+
+#Path to the to-be-created handler file specific to the task
+handlerPath="$WORKDIR/scripts/mobilevit_handler.py"
+
+#Create the specific handler .py file for the task at specified path
+cd $ROOT/serve/examples/Huggingface_Transformers && chmod +x create_hf_handler.sh && ./create_hf_handler.sh -t $task -f $framework -o $handlerPath
+
+#Download the model from HF hub and create model archive
+cd $ROOT/serve/examples/Huggingface_Transformers && chmod +x prepare_mar_from_hf.sh && ./prepare_mar_from_hf.sh -p $handlerPath -n $modelName -d $WORKDIR -u $repoUrl
+
+cd $WORKDIR
+```
+
+* After waiting for a few seconds for the server to boot up, register the downloaded model on the Torchserve server:
+```
+curl -X POST "localhost:8081/models?url=$modelName.mar&batch_size=8&max_batch_delay=10&initial_workers=1"
+```
+
+* Refer to the `Managing registered models` and `Inferencing and Benchmarking` sections below for auxillary details.
+
+## Setup
+Install the model archiver related dependencies:
+```
+pip install torch-model-archiver
+```
+
+Follow the [Docker installation guide](https://docs.docker.com/engine/install/) in case it is not installed.
+
+## Running the Torchserve server
+Cloning the repo: 
+```
+ROOT=$(pwd)
+git clone https://github.com/pytorch/serve.git
+```
+
+Building a Torchserve CPU Docker container (or GPU/IPEX containers following the [original guide](https://github.com/pytorch/serve/tree/master/docker#create-torchserve-docker-image)):  
+```
+cd $ROOT/serve/docker && ./build_image.sh -bt production -t torchserve-cpu-prod && cd $ROOT
+```
+
+Checking whether the Torchserve image is present in the list of Docker images:
+```
+docker images
+```
+
+Running the Torchserve server container with Docker while binding Model store & HF model directories (refer to [this](https://github.com/pytorch/serve/tree/master/docker#create-torch-model-archiver-from-container) for more details):
+
+```
+WORKDIR="$ROOT/serve/examples/Huggingface_Transformers/Image_classification_docker"
+cd $WORKDIR && mkdir -p HF-models && mkdir -p model-store
+docker run -d --rm -it --shm-size=50g -p 8080:8080 -p 8081:8081 --name torchserve-cpu-prod --mount type=bind,source=$WORKDIR/scripts/config.properties,target=/home/model-server/config.properties --mount type=bind,source=$WORKDIR/model-store,target=/home/model-server/model-store --mount type=bind,source=$WORKDIR/HF-models,target=/home/model-server/HF-models torchserve-cpu-prod torchserve --ncs --model-store=/home/model-server/model-store --ts-config /home/model-server/config.properties
+```
+
+Checking whether the server was started properly after waiting for a few seconds:
+```
+curl http://127.0.0.1:8080/ping
+#OR
+curl http://127.0.0.1:8081/models/
+```
+
+Installing git-lfs to be able to download 🤗 models from the hub:
+```
+pip install git-lfs
+```
+
+## Registering a 🤗 model from the [hub](https://huggingface.co/models)
```

Comment:
Ideally it would make sense to write both steps for Docker and normal Torchserve as written in various parts of the docs. 

I haven't experimented this yet, but I'm not sure whether with `torchserve --start` the handler files would be able to access additional folders like `HF-models`, due to which we are using Docker container to load locally saved HF models with `pipeline` without making a copy of them (done via shared volumes with the container), which has two advantages:
(i) The HF models can be saved in a designated storage without wasting bandwidth (by downloading from Huggingface) every time a new Torchserve Huggingface container is to be made. For example: The `Image_classification_docker` example currently might be used conveniently with AWS ELB to first fetch the Docker image and downloading HF models saved on S3 bucket to be loaded by Huggingface pipeline.  
(ii) This also avoids making a copy of the model checkpoint in storage which generally happens while creating a `mar`, this could crucial storage optimization for deploying LLMs.  

I'd like to find out what it would have been without docker to implement the same flow, but I currently can't afford the time to do so.